frontend/components/YourWalletPage/WithdrawFunds.tsx

```diff
@@ -0,0 +1,141 @@
+import { Button, Flex, Input, message, Modal, Typography } from 'antd';
+import { isAddress } from 'ethers/lib/utils';
+import React, { useCallback, useState } from 'react';
+
+import { useBalance } from '@/hooks/useBalance';
+import { useServices } from '@/hooks/useServices';
+import { useWallet } from '@/hooks/useWallet';
+import { ServicesService } from '@/service/Services';
+import { Address } from '@/types/Address';
+
+import { CustomAlert } from '../Alert';
+
+const { Text } = Typography;
+
+export const WithdrawFunds = () => {
+  const { updateWallets } = useWallet();
+  const { updateBalances } = useBalance();
+  const { service, updateServicesState } = useServices();
+  const serviceHash = service?.hash;
+
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [withdrawAddress, setWithdrawAddress] = useState('');
+  const [isWithdrawalLoading, setIsWithdrawalLoading] = useState(false);
+
+  const showModal = useCallback(() => {
+    setIsModalVisible(true);
+  }, []);
+
+  const handleCancel = useCallback(() => {
+    setIsModalVisible(false);
+    setWithdrawAddress('');
+  }, []);
+
+  const refetchDetails = useCallback(async () => {
+    try {
+      await updateServicesState();
+      await updateWallets();
+      await updateBalances();
+    } catch (error) {
+      console.error('Failed to refetch details after withdrawal', error);
+    }
+  }, [updateServicesState, updateWallets, updateBalances]);
+
+  const handleProceed = useCallback(async () => {
+    if (!withdrawAddress) return;
+    if (!serviceHash) return;
+
+    const isValidAddress = isAddress(withdrawAddress);
+    if (!isValidAddress) {
+      message.error('Please enter a valid address');
+      return;
+    }
+
+    setIsWithdrawalLoading(true);
+    message.loading('Withdrawal pending. It may take a few minutes.');
+
+    try {
+      const response = await ServicesService.withdrawBalance({
+        withdrawAddress: withdrawAddress as Address,
+        serviceHash: serviceHash,
+      });
```

Comment:
this probably needs validation to check user's can withdraw..
some funds are locked in staking contracts for X time
not sure if critical for now, however

Reply:
can use this one `isServiceStakedForMinimumDuration` to hide the button if it's false, could be enough? @mohandast52 @truemiller